test/e2e/basic_test.go

```diff
@@ -263,10 +272,10 @@ func TestIPAMRestart(t *testing.T) {
 	if podIP2, err = createPodAndGetIP(podName2); err != nil {
 		t.Fatalf("Failed to retrieve IP for Pod '%s': %v", podName2, err)
 	}
-	t.Logf("Pod '%s' has IP address %s", podName2, podIP2)
+	t.Logf("Pod '%s' has IP addresses %v", podName2, podIP2)
 
-	if podIP1 == podIP2 {
-		t.Errorf("Pods '%s' and '%s' were assigned the same IP %s", podName1, podName2, podIP1)
+	if podIP1.hasSameIP(podIP2) {
+		t.Errorf("Pods '%s' and '%s' were assigned the same IP %v v.s. %v", podName1, podName2, podIP1, podIP2)
```

Comment:
If both IPs are the same, why do we need to print both in the error message?

Reply:
removed.